### PR TITLE
refactor: annotation key validation should be unified

### PR DESCRIPTION
**Pull request by pandaamanda**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

annotation key support uppercase letter，validate logic should  be unified.
https://github.com/kubernetes/kubernetes/blob/f173d01c011c3574dea73a6fa3e20b0ab94531bb/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L46-L53

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go [hunk 1]
```diff
@@ -910,6 +910,7 @@ func NewMutatingTestCases(url *url.URL, configurationName string) []MutatingTest
 				"mutation.webhook.admission.k8s.io/round_0_index_1": mutationAnnotationValue(configurationName, "removeLabel", true),
 				"patch.webhook.admission.k8s.io/round_0_index_0":    patchAnnotationValue(configurationName, "addLabel", `[{"op": "add", "path": "/metadata/labels/added", "value": "test"}]`),
 				"patch.webhook.admission.k8s.io/round_0_index_1":    patchAnnotationValue(configurationName, "removeLabel", `[{"op": "remove", "path": "/metadata/labels/remove"}]`),
+				"removeLabel/key1": "value1",
```
Comment (roycaihw) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go:913]: Do you know who set this label? Was it failing silently before this change?

File: staging/src/k8s.io/apiserver/pkg/admission/attributes.go [hunk 2]
```diff
@@ -204,7 +204,7 @@ func checkKeyFormat(key string) error {
 	if len(parts) != 2 {
 		return fmt.Errorf("annotation key has invalid format, the right format is a DNS subdomain prefix and '/' and key name. (e.g. 'podsecuritypolicy.admission.k8s.io/admit-policy')")
 	}
-	if msgs := validation.IsQualifiedName(key); len(msgs) != 0 {
+	if msgs := validation.IsQualifiedName(strings.ToLower(key)); len(msgs) != 0 {
```
Comment (roycaihw) [inline staging/src/k8s.io/apiserver/pkg/admission/attributes.go:207]: The validation in metadata.annotation is a little confusing today https://github.com/kubernetes/kubernetes/pull/109892. Not sure if we should change the validation in audit annotation to match that, or simply improve our documentation.

Comment (pandaamanda) [reply staging/src/k8s.io/apiserver/pkg/admission/attributes.go:207]: ok, i will close it. 
tks

**Pull request closed without merging**